### ISSUE TITLE
Bug: error when using nix-shell 1.11.1 as script interpreter

### ISSUE THREAD

**Issue by FRidh** [NixOS/nix#790]
Trying to run any of the fragments shown in https://github.com/NixOS/nixpkgs/issues/12798 fails with both
- `nix-shell script` 
- `./script`

It fails with
```
Use of uninitialized value $line in substitution (s///) at /nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/lib/perl5/5.20.3/Text/ParseWords.pm line 20, <SCRIPT> line 2.
Use of uninitialized value $line in substitution (s///) at /nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/lib/perl5/5.20.3/Text/ParseWords.pm line 20, <SCRIPT> line 3.
error: getting status of ‘/tmp/1’: No such file or directory
```
The fragments work without issues with `nix-shell 1.10` (15.09).

**Comment by edolstra** [comment]
Fixed by 3baf8be1d149e034f771d2a0acf9d50563164cf6.

**Comment by johan-boule** [comment]
I'm wondering which channel contains this fix.
